### DISCUSSION TITLE
More Intuitve Mouse Controls

### DISCUSSION THREAD

**Discussion by SirNate0** [pyvista/pyvista#5463] [Ideas]
I'm not certain the mechanism behind it, but I think trying to have mouse controls that act more like Blender, for example, may give a better user experience. As it is, the mouse controls cause some fairly non-intiutive behavior:

For example, if you move the mouse in a full circle around the plot (with the left button held), the model on the screen will not rotate back to it's original position, but to some rotation the model's up is no longer in the up direction on the screen. If you keep moving in the circle, it seems it eventually returns to the initial rotation several circles later (number depending on the radius of your circle on the screen, I think). I find this behavior to be unintuitive and difficult to use - I generally want Up to remain Up and not Off-to-the-Side, so after moving the model to view it's top or bottom, when I use the mouse to move it back to a side view I still expect Up on the model to match Up on my screen.

**Comment by MatthewFlamm** [comment]
There might be an available interactor that gets close to what you want, maybe `enable_terrain_style`?  See [docs](https://docs.pyvista.org/version/stable/api/plotting/_autosummary/pyvista.plotter.enable_terrain_style). If there isn't an interactor style that has a mode you need, you'd have to code the mouse movement logic yourself.  This vtk code should be a good starting point as it has separate elevation and azimuthal rotation calculations.  [code for Rotate](https://github.com/Kitware/VTK/blob/f9fead586e12dd131fab60e884785aeeeafe58f9/Interaction/Style/vtkInteractorStyleTrackballCamera.cxx#L237-L273).

In the upcoming release, there will be customizations for the default style to assign different mouse actions, but only from within the set specified already by vtk.  We could consider opening this customization to all available interactor styles.  This is helpful if the action is already available but you want to remap to another mouse button.

**Reply by SirNate0** [reply]
Thank you, yes, that interactor is basically what I want.
